test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -122,6 +122,19 @@ public void AutoFitRegressionTest()
 
         [LightGBMFact]
         public void AutoFitRankingTest()
+        {
+            // Testing Ranking with both TrainValidateRunner and CrossValSummaryRunner
+            // with corresponding baseline metrics. The metrics of Ranking with
+            // CrossValSummaryRunner is lower due to the 10 cross validation folds
+            // and due to the fact that CrossValSummaryRunner is used only when
+            // fewer than <1500 rows of training data is given.
+            AutoFitRankingTestTemplate(true, 0.4, 20);
+            AutoFitRankingTestTemplate(false, 0.13, 9.7);
+            AutoFitRankingCVTestTemplate(true, 0.4, 19);
+            AutoFitRankingCVTestTemplate(false, 0.13, 9.7);
```

Comment:
Is the drop in score between the TrainValidate & CV runner due to the smaller dataset? Might be worth a run without subsampling to see that the scores are pretty close between the two.

Reply:
The lower benchmark metrics is due to the lower provided accuracies in this line, where no validation data is being provided:
https://github.com/dotnet/machinelearning/blob/ef8e085a194ba6e8ca942ea464381d589acc37a0/test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs#L160
As CV depends on validation data, this exact experiment may not be applicable for when running with `CrossValSummaryRunner`. @justinormont thoughts?

Reply:
@justinormont The metric results in the experiments below are differing significantly. Specifically, the fist experiment has a lot lower metrics than the other experiments:

https://github.com/dotnet/machinelearning/blob/5370692f6d1e6f1f686596d530032b4226dc9e80/test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs#L143-L160

Do you know why this is happening? Also, do you know why we are enforcing `SamplingKeyColumnName` and `GroupIdColumnName` to have the same column names for ranking?

https://github.com/dotnet/machinelearning/blob/5370692f6d1e6f1f686596d530032b4226dc9e80/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L67-L89

Reply:
SamplingKeys ensure that samples that have the same SamplingKey get grouped together within the training set and the test set. Ranking needs to have samples that have the same GroupId to be grouped together, so you can't have a sample with groupId A be both in the training set and in the test set. That means we need to enforce the samplingkey to be the GroupId only for ranking CV experiments.

Reply:
Gotcha, thanks Keren!